### PR TITLE
IDEMPIERE-6016:test case to simulate transaction timeouts and ensure no open connections remain afterwards

### PR DESCRIPTION
**Pull request by hieplq**
To observe the connection leak, please revert commit 6acc6f31 before running this test case.

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.idempiere.test/src/org/idempiere/test/base/DBTest.java [hunk 1]
```diff
@@ -373,4 +377,47 @@ else if (DB.isOracle())
...
+			Thread.sleep(40000);//Wait for the transaction monitor to complete its task
+			
+			int afterActiveConnection = getNumConnectPerStatus(DB.getDatabase().getStatus(), REG_ACTIVE_CONNECT);
+			assertEquals(beforeActiveConnection, afterActiveConnection);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+
```
Comment (hengsin) [inline org.idempiere.test/src/org/idempiere/test/base/DBTest.java:422]: this needs to be executed in isolation from other test. to ensure that, add `@Isolated` and `@Execution(ExecutionMode.SAME_THREAD)` annotation to the DBTest class.

**Comment by hengsin** [comment]
[pr2219.patch.txt](https://github.com/idempiere/idempiere/files/14110798/pr2219.patch.txt)
- replace abort with rollback
- use shorter duration for timeout test 2
- add Isolatead and same thread execution class annotation

**Review by hengsin — CHANGES_REQUESTED** [review]
hi @hieplq , please review attached patch, thanks.

**Comment by hieplq** [comment]
>     * replace abort with rollback

Modifying this code, will the timeout now fire the AfterRollbackEvent instead of its original behavior?

**Comment by hengsin** [comment]
> > ```
> > * replace abort with rollback
> > ```
> 
> Modifying this code, will the timeout now fire the AfterRollbackEvent instead of its original behavior?

No, to fire AfterRollbackEvent, you need to call Trx.rollback instead of Connection.rollback.

**Comment by hieplq** [comment]
we need to call setAutoCommit (true) also?

**Comment by hengsin** [comment]
> we need to call setAutoCommit (true) also?

You means to duplicate what's in Trx.close ? Well, perhaps it is safer to do that as well. I guess we can try to just call the rollback and close method of Trx instead.

**Comment by hieplq** [comment]
> You means to duplicate what's in Trx.close ? 

yeb

> Well, perhaps it is safer to do that as well. I guess we can try to just call the rollback and close method of Trx instead.

This triggers the event, so timeouts case might require overloaded rollback/close methods to no raise event

**Comment by hengsin** [comment]
hi @hieplq, I don't think any triggering of trx event (rollback, close) is a problem here.  Also, this is suppose to happens rarely and don't think we should put too much design effect into this.

**Comment by hengsin** [comment]
hi @hieplq , I try with simplifying that to use existing trx.rollback and trx.close method and it seems work just fine.

[pr2219.patch1.txt](https://github.com/idempiere/idempiere/files/14174190/pr2219.patch1.txt)

**Comment by hieplq** [comment]
it's ok for me

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**